packages/session-replay-browser/src/session-replay.ts

```diff
@@ -204,6 +204,15 @@ export class SessionReplay implements AmplitudeSessionReplay {
   }
 
   getBlockSelectors(): string | string[] | undefined {
+    // For some reason, this defaults to empty array ([]) if undefined in the compiled script.
+    // Empty arrays cause errors when being evaluated in Safari.
+    // Force the selector to be undefined if it's an empty array.
+    if (
+      Array.isArray(this.config?.privacyConfig?.blockSelector) &&
+      this.config?.privacyConfig?.blockSelector.length === 0
+    ) {
+      return undefined;
+    }
```

Comment:
NIt: I'd rework this slightly.
```suggestion
    const blockSelector = this.config?.privacyConfig?.blockSelector ?? [];
    if (blockSelector.length === 0) {
      return undefined;
    }
    return blockSelector;
```

This is mostly because the JS generated has a smaller footprint: https://gist.github.com/lewgordon-amplitude/b84c69d0518053ee68ed90218a7c349b

Reply:
good point. Was concerned because block selector could be string, but `.length` also works on strings =)